lib/jerboa/format/body/attribute/lifetime.ex

```diff
@@ -0,0 +1,37 @@
+defmodule Jerboa.Format.Body.Attribute.Lifetime do
+  @moduledoc """
+  LIFETIME attribute as defined in [TURN RFC](https://trac.tools.ietf.org/html/rfc5766#section-14.2)
+  """
+
+  alias Jerboa.Format.Body.Attribute
+  alias Jerboa.Format.Lifetime.LengthError
+
+  defstruct duration: 0
+
+  @max_duration :math.pow(2, 32) - 1
+
+  @typedoc """
+  Represents a lifetime of the allocation
+
+  * `:duration` is a duration of a lifetime in seconds
+  """
+  @type t :: %__MODULE__{
+    duration: non_neg_integer
+  }
+
+  @doc false
+  @spec encode(t) :: binary
+  def encode(%__MODULE__{duration: duration})
+    when is_integer(duration) and duration <= @max_duration and duration >= 0 do
```

Comment:
the guard can be reduced to `when duration in 0..@max_druation` - if it's not integer that you will get the same error: `FuncionCaluse`